modules: memfault: Add support for uploading modem traces to Memfault (#21252)
nrfconnect/sdk-nrf: samples/debug/memfault/README.rst

   :start-after: modem_lib_sending_traces_UART_start
   :end-before: modem_lib_sending_traces_UART_end

Automatic capture and upload modem traces with coredumps
--------------------------------------------------------

> The sample can be configured to upload modem traces to Memfault when a coredump is triggered.
> To enable this feature, include the following overlay in your project:
> 
> * :file:`overlay-modem_trace_to_memfault.conf`

Reviewer: ```suggestion
You can configure the sample to upload modem traces to Memfault when a coredump is triggered.
To enable this feature, include the :file:`overlay-modem_trace_to_memfault.conf` overlay in your project.
```

> .. include:: libraries/debug/memfault_ncs.rst

Reviewer: ```suggestion
.. include:: /libraries/debug/memfault_ncs.rst
```
This should fix the doc build failure.
